core/src/test/java/org/elasticsearch/index/query/AbstractQueryTestCase.java

```diff
@@ -472,8 +487,21 @@ public void testEqualsAndHashcode() throws IOException {
         assertTrue("equals is not symmetric", thirdQuery.equals(secondQuery));
         assertTrue("equals is not symmetric", thirdQuery.equals(firstQuery));
 
+        // Now call toQuery on the secondQuery and make sure its still equal to
+        // firstQuery
+        QueryShardContext context = createShardContext();
+        secondQuery.toQuery(context);
+        assertTrue("query is not equal to its copy after calling toQuery, firstQuery: " + firstQuery + ", secondQuery: " + secondQuery,
+                firstQuery.equals(secondQuery));
+        assertTrue("equals is not symmetric after calling toQuery, firstQuery: " + firstQuery + ", secondQuery: " + secondQuery,
+                secondQuery.equals(firstQuery));
+        assertThat("query copy's hashcode is different from original hashcode after calling toQuery, firstQuery: " + firstQuery
+                + ", secondQuery: " + secondQuery, secondQuery.hashCode(),
+                equalTo(firstQuery.hashCode()));
+
```

Comment:
not sure about this, I remember trying to add something like this, but having to back off due to test failures. Do all query tests pass after this change?